src/core/mac/mac_frame.hpp

```diff
@@ -1107,6 +1081,48 @@ class RxFrame : public Frame
 class TxFrame : public Frame
 {
 public:
+    /**
+     * Represents header information.
+     *
+     */
+    struct Info : public Clearable<Info>
+    {
+        /**
+         * Initializes the `Info` by clearing all its fields (setting all bytes to zero).
+         *
+         */
+        Info(void) { Clear(); }
+
+        /**
+         * Prepares MAC headers based on `Info` fields in a given `TxFrame`.
+         *
+         * This method uses the `Info` structure to construct the MAC address and security headers in @p aTxFrame.
+         * It determines the Frame Control Field (FCF), including setting the appropriate frame type, security level,
+         * and addressing mode flags. It populates the source and destination addresses and PAN IDs within the MAC
+         * header based on the information provided in the `Info` structure.
+         *
+         * It sets the Ack Request bit in the FCF if the following criteria are met:
+         *   - A destination address is present
+         *   - The destination address is not the broadcast address
+         *   - The frame type is not an ACK frame
+         *
+         * The Frame Pending and IE Present flags in the FCF are not set. They may need to be set separately depending
+         * on the specific requirements of the frame being transmitted.
+         *
+         * @param[in] aTxFrame  The `TxFrame` whose MAC headers are to be prepared.
+         *
+         */
+        void PrepareHeadersIn(TxFrame &aTxFrame) const;
```

Comment:
Will `PrepareHeadersFor` be better? The current one also works for me.

Reply:
Kept it as `PrepareHeadersIn()`. I like the use of `In` term to indicate that the passed-in `TxFrame` is being updated.